internal/payloads/validation/id.go

```diff
@@ -0,0 +1,26 @@
+package validation
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+var (
+	idValidation = regexp.MustCompile("^[0-9]+$")
+	ErrInvalidId = fmt.Errorf("invalid id")
+)
+
+func DigitsOnly(id string) error {
+	if !idValidation.MatchString(id) {
+		return ErrInvalidId
+	}
+
+	// Checking for out of range error
+	_, err := strconv.ParseInt(id, 10, 64)
+	if err != nil {
+		return ErrInvalidId
+	}
+
+	return nil
+}
```

Comment:
I still do not understand why there is a regular expression needed, both `strconv.Atoi` and `strconv.ParseInt` will return an error in case the string does not contain a valid number: https://go.dev/play/p/XFoPrKfctrC

On top of that, the proposed regular expression will not work for negative numbers.

Reply:
I was afraid about how Golang would handle numbers from other alphabets since we never really know what things from libraries do under the hood.
I thought that the regex would not hurt, and wanted to make sure, that some exotic things like Persian numbers would not pass the validation. But I have added some more values into the test before removing the regex, so it should be hopefully okay.